src/python/pants/backend/project_info/rules/dependencies_integration_test.py

```diff
@@ -25,11 +33,26 @@ def create_python_library(self, path: str, *, dependencies: Optional[List[str]]
             path=path, target_type="python_library", name="target", dependencies=dependencies or []
         )
 
+    def create_python_requirement_library(self, name: str) -> None:
+        self.create_library(
+            path=f"3rdparty/{name}",
+            target_type="python_requirement_library",
+            name=name,
+            requirements=f"[python_requirement('{name}==1.0.0')]",
+        )
+
     def assert_dependencies(
-        self, *, target: str, expected: List[str], transitive: bool = True
+        self,
+        *,
+        target: str,
+        expected: List[str],
+        transitive: bool = True,
+        dependency_type: str = "source",
```

Comment:
I recommend using the enum `DependencyType` so that we're working with a constrained set of symbols, rather than strings. For example, it reduces the risk of accidentally using the string `sources`. 

See https://github.com/pantsbuild/pants/pull/9258/files#diff-b614bf64c449e29cad46598173047cff for an example.